### ISSUE TITLE
Implement "grid fit view" similar to FontForge

### ISSUE THREAD

**Issue by madig** [trufont/trufont#171]
A glyph view overlay of sorts that takes a point-size and a DPI value and paints a live pixel grid with pixel coverage over the glyph, allowing precise outline placing to optimize pixel coverage of the outline. Taking hinting into account is probably much more complicated, and I think an unhinted view will do at first.

**Comment by anthrotype** [comment]
nice idea!
If we use the FreeType's Python bindings ([freetype-py](https://github.com/rougier/freetype-py/)), "taking hinting into account" is not much more complicated, but comes for free.
Btw, that's the same that the TTH Robofont extension is using.

**Comment by anthrotype** [comment]
well, let me rephrase that. If you have an already compiled binary font, that's easy. It you want the glyph view to update "live", well that's another story..

**Comment by madig** [comment]
Hm. Maybe the glyph can be compiled as a single-glyph font after each change and written to memory, so FreeType can be fed a pointer or something? That would probably throw off the autohinter, though. And it would require an in-memory font compiler.

**Comment by adrientetar** [comment]
> It you want the glyph view to update "live", well that's another story..

It's perfectly doable with fontTools, you just need to create a mini-font as is done in TTH.

Btw, I don't know if such functionality will be part of the main editor or put to extensions but there will be a way to generate fonts and/or rasterize individual glyphs with FT easily. I personally want to make a bridge with harfbuzz and with Qt WebEngine Widgets, both of which require compiling fonts fast, which I think is perfectly doable (as in FontForge).

**Comment by adrientetar** [comment]
![bitmap](https://cloud.githubusercontent.com/assets/1250894/12007331/c2577db2-abff-11e5-8f70-8dd0a6aceff7.PNG)

**Comment by madig** [comment]
Nice :) is the outline unhinted? This would be a nice candidate for gamma correction, btw :grin:

**Comment by adrientetar** [comment]
It's freetype's default mode. I'll look at this more when the toolchain is ready.

**Comment by madig** [comment]
Then it probably has some sort of hinting turned on.

By the way, bonus points for you if you implement a subpixel rendering view like on http://www.beatstamm.com/typography/RTRCh3.htm ;)

**Comment by adrientetar** [comment]
Not in the main app.
